cmds/fmap/fmap.go

```diff
@@ -64,11 +77,24 @@ func printFMap(f *fmap.FMap, m *fmap.FMapMetadata) {
 	}
 }
 
+func btoi(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
 func main() {
 	flag.Parse()
 
-	if *summary && *read >= 0 {
-		log.Fatal("Both flags cannot be used at once")
+	// Validate flags
+	if btoi(*summary)+btoi(*read >= 0)+btoi(*checksum != "") > 1 {
```

Comment:
There are a few limitations with the standard `flag` package. My pet peeves are:

- You always have to set a default value and it cannot be nil. However, it is still possible to check if a flag appeared in the command line via `flag.Visit`, but it is not straightforward.
- No contractions for boolean flags, so `tar -zxf FILE` has to be written as `tar -z -x -f FILE`.
- Making aliases such as `-v` == `--verbose` is not as easy at is should be.

There are some lightweight, gnu-like parsers out there. It might be worth switching.

Anyways, I will definitely add the `var btoi = map[bool] int{false: 0, true:1, }`!